plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcMetadataFactory.java

```diff
@@ -36,6 +32,6 @@ public JdbcMetadataFactory(JdbcClient jdbcClient, JdbcMetadataConfig config)
 
     public JdbcMetadata create()
     {
-        return new JdbcMetadata(new TransactionCachingJdbcClient(jdbcClient, new Duration(1, TimeUnit.DAYS)), allowDropTable);
+        return new JdbcMetadata(jdbcClient, allowDropTable);
```

Comment:
Even if caching is generally disabled, we should still have a `CachingJdbcClient` instance here for in-transaction cache

Reply:
What if jdbcClient already is CachingJdbcClient?

Reply:
This could be used to check if we have caching jdbc client already https://github.com/trinodb/trino/pull/7470

Reply:
> What if jdbcClient already is CachingJdbcClient?

doesn't matter.
we want strong caching without invalidation within a transaction (like L1 cache), and usual invalidation for a shared cache (like L2 cache)

Reply:
Sure. Done.